controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -173,102 +212,244 @@ func (r *ClowdJobInvocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
+		}
+	}
+	envConfig["cdappconfigs"] = appConfigs
+
+	// Marshall the data into the top level "cdenvconfig.json" to be mounted as a single secret
+	envData, err := json.Marshal(envConfig)
+	if err != nil {
+		r.Log.Error(err, "Failed to marshal iqe secret")
+		return err
+	}
+
+	cdEnv := make(map[string][]byte)
+	cdEnv["cdenvconfig.json"] = envData
+
+	iqeSecret.Data = cdEnv
+	if err := update.Apply(ctx, r.Client, iqeSecret); err != nil {
+		r.Log.Error(err, "Failed to create iqe secret")
+		return err
+	}
+
+	return nil
+}
+
+func (r *ClowdJobInvocationReconciler) createIqeJobResource(cji *crd.ClowdJobInvocation, env *crd.ClowdEnvironment, app *crd.ClowdApp, nn types.NamespacedName, ctx context.Context, j *batchv1.Job) {
 	labels := cji.GetLabels()
 	cji.SetObjectMeta(j, crd.Name(nn.Name), crd.Labels(labels))
 
 	j.ObjectMeta.Labels = labels
 	j.Spec.Template.ObjectMeta.Labels = labels
 
-	pod := job.PodSpec
-
-	if job.RestartPolicy == "" {
-		j.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
-	} else {
-		j.Spec.Template.Spec.RestartPolicy = job.RestartPolicy
-	}
+	j.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
 
 	j.Spec.Template.Spec.ImagePullSecrets = []core.LocalObjectReference{
 		{Name: "quay-cloudservices-pull"},
 	}
 
-	envvar := pod.Env
-	envvar = append(envvar, core.EnvVar{Name: "ACG_CONFIG", Value: "/cdapp/cdappconfig.json"})
-
-	var livenessProbe core.Probe
-	var readinessProbe core.Probe
+	pod := crd.PodSpec{
+		Resources: env.Spec.Providers.Iqe.Resources,
+	}
 
-	if pod.LivenessProbe != nil {
-		livenessProbe = *pod.LivenessProbe
+	envvar := []core.EnvVar{}
+	envvar = append(envvar, core.EnvVar{Name: "ACG_CONFIG", Value: "/cdapp/cdappconfig.json"})
+	envvar = append(envvar, core.EnvVar{Name: "ENV_FOR_DYNACONF", Value: cji.Spec.Iqe.DynaconfEnvName})
+	envvar = append(envvar, core.EnvVar{Name: "NAMESPACE", Value: nn.Namespace})
+	envvar = append(envvar, core.EnvVar{Name: "CLOWDER_ENABLED", Value: "true"})
+	// envvar = append(envvar, core.EnvVar{Name: "IQE_TESTS_LOCAL_CONF_PATH", Value: "/iqe_settings"})
+
+	tag := ""
+	if cji.Spec.Iqe.ImageTag != "" {
+		tag = cji.Spec.Iqe.ImageTag
 	} else {
-		livenessProbe = core.Probe{}
+		tag = app.Spec.Iqe.Plugin
 	}
-	if pod.ReadinessProbe != nil {
-		readinessProbe = *pod.ReadinessProbe
-	} else {
-		readinessProbe = core.Probe{}
+	plugin := app.Spec.Iqe.Plugin
+	iqeImage := env.Spec.Providers.Iqe.ImageBase
+
+	accessLevel := env.Spec.Providers.Iqe.K8SAccessLevel
+
+	// TODO: implement access level with service account
+	switch accessLevel {
+	// Use edit level service account to create and delete resources
+	// one per app when the app is created
+	case "edit":
+		// TODO: Create secret here
+		j.Spec.Template.Spec.ServiceAccountName = fmt.Sprintf("%v-iqe", app.Name)
+	// Standard view access to the owned resources
+	case "view":
+		j.Spec.Template.Spec.ServiceAccountName = fmt.Sprintf("%v", app.Name)
+	// Standard access
+	case "default":
```

Comment:
case default - don't we just omit the service account name entirely?

Reply:
Yeah, that's probably correct. I wasn't sure if the default behavior was to use the `default` service account or not.